app/model/block.py

```diff
@@ -10,11 +10,73 @@ def __init__(self):
         self.container = None
         self.questionnaire = None
         self.validation = None
-        self.questionniare = None
+        self.questionnaire = None
         self.templatable_properties = []
         self.display = Display()
 
     def add_section(self, section):
         if section not in self.sections:
             self.sections.append(section)
             section.container = self
+
+    def to_json(self):
+        json_dict = {
+            "id": self.id,
+            "title": self.title,
+            "sections": [],
+            "validation": [],
+            "display": {}
+        }
+
+        for section in self.sections:
+            json_dict['sections'].append(section.to_json())
+
+        if self.validation is not None:
+            for validation in self.validation:
+                json_dict['validation'].append(validation.to_json())
+
+        if self.display is not None:
+            json_dict['display'] = self.display.to_json()
+
+        return json_dict
+
+    def __eq__(self, other):
```

Comment:
You've got to be careful with this, anywhere you'd normally use a == (or code that performs the same comparison) is going to call this code and not compare the object pointers; which may not be obvious to the caller and give strange behaviour if not expected, if you do stick with it then you probably need pretty good unit tests around the operator to ensure it is working correctly (i.e. verifies each change in property results in a not equal outcome etc.). This seems like a lot of code to create and maintain just to test the json stuff; do you really want to do/keep these methods, another approach would be to run this through the to_json method, dump the resulting JSON text and paste that text into the unit test to verify that the code generates the expected json text or manually in the unit test check the properties of the objects etc. Worth discussing with @dhilton @LJBabbage ?